plugins/woocommerce/src/Internal/AddressProvider/AbstractAutomatticAddressProvider.php

```diff
@@ -0,0 +1,165 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Internal\AddressProvider;
+
+use Automattic\WooCommerce\StoreApi\Utilities\JsonWebToken;
+use Automattic\Jetpack\Constants;
+use WC_Address_Provider;
+
+/**
+ * Abstract Automattic address provider is an abstract implementation of the WC_Address_Provider that is meant to be used by Automattic services to get support for address autocomplete and maps with minimal code maintenance.
+ *
+ * @since 10.1.0
+ * @package WooCommerce
+ */
+abstract class AbstractAutomatticAddressProvider extends WC_Address_Provider {
+
+	/**
+	 * The JWT for the address service.
+	 *
+	 * @var string
+	 */
+	private $jwt = null;
+
+	/**
+	 * Loads up the JWT for the address service and saves it to transient.
+	 */
+	public function __construct() {
+		add_filter( 'pre_update_option_woocommerce_address_autocomplete_enabled', array( $this, 'refresh_cache' ) );
```

Comment:
Every time the address autocomplete feature get enabled or disabled, we refresh the cache, just in case we have malformed cache.